File: go/raindrops/raindrops.go
package raindrops

import (
    "strconv"
)

const testVersion = 3

// Convert converts integer into string
func Convert(x int) string {
    result := ""
    if x % 3 == 0 {
        result += "Pling"
    }
    if x % 5 == 0 {
        result += "Plang"
    }
    if x % 7 == 0 {
        result += "Plong"
    }
    if result == "" {
        return strconv.Itoa(x)
    } else {
        return result
    }
}

// Don't forget the test program has a benchmark too.
// How fast does your Convert convert?

File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"

namespace raindrops {

    std::string convert(int n) {
        std::string s = "";
        if (n % 3 == 0) {
            s += "Pling";
        }
        if (n % 5 == 0) {
            s += "Plang";
        }
        if (n % 7 == 0) {
            s += "Plong";
        }
        if (s.length() > 0) {
            return s;
        }
        return std::to_string(n);
    }

}
